### PR TITLE
New formula: ICC for Mac v1.0.r7564

### PR DESCRIPTION
**Pull request by apetresc**
ICC is the most popular chess server worldwide. This is their official Mac client.

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by apetresc** [comment]
Okay, yeah, I wasn't sure about the "for Mac" part of the name. I opted to keep it because it's definitely a part of the app's own branding; the .app file is called "ICC for Mac.app", etc. But either way :)

**Comment by apetresc** [comment]
Apparently Travis is failing on `#{version.patch}` not resolving to anything. Does it not work with non-numerical patch numbers? (the 'r' in this case?)

**Comment by apetresc** [comment]
Indeed, that does seem to be the problem. The [current `patch` regex](https://github.com/Homebrew/brew/blob/19347f3b87917368d7157fe5007d506a4db0c7db/Library/Homebrew/cask/lib/hbc/dsl/version.rb#L13) only works for purely-numerical patch numbers.

**Review by vitorgalvao — APPROVED** [review]

File: Casks/icc.rb [hunk 1]
```diff
@@ -9,4 +9,11 @@
   pkg "ICCforMac.#{version.after_comma}.pkg"
 
   uninstall pkgutil: 'com.chessclub.*'
+
+  zap delete: '~/Library/Saved Application State/com.chessclub.desktop-icc.savedState',
+      trash:  [
+                '~/.cache/internet_chess_club',
```
Comment (vitorgalvao) [inline Casks/icc.rb]: Caches are for `delete`.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Pull request merged**